### ISSUE TITLE
Throughput with JITServer dropped 2.4% between 08/04 and 08/05

### ISSUE THREAD

**Issue by mpirvu** [eclipse/openj9#10355]
I measured AcmeAir throughput with JITServer turned on and observed a 2.4% throughput drop between these two OpenJ9 builds:

```
mpirvu@skylake:/opt/IBM/AcmeAir/scripts$ /home/mpirvu/sdks/OpenJDK8U-jre_x64_linux_openj9_2020-08-04-03-59/bin/java -version
openjdk version "1.8.0_265"
OpenJDK Runtime Environment (build 1.8.0_265-202008031718-b01)
Eclipse OpenJ9 VM (build master-1c194db6e, JRE 1.8.0 Linux amd64-64-Bit Compressed References 20200803_733 (JIT enabled, AOT enabled)
OpenJ9   - 1c194db6e
OMR      - 7b76a160b
JCL      - bf6e4d1c based on jdk8u265-b01)
```
```
mpirvu@skylake:/opt/IBM/AcmeAir/scripts$ /home/mpirvu/sdks/OpenJDK8U-jre_x64_linux_openj9_2020-08-05-08-20/bin/java -version
openjdk version "1.8.0_265"
OpenJDK Runtime Environment (build 1.8.0_265-202008041719-b01)
Eclipse OpenJ9 VM (build master-2020312e3, JRE 1.8.0 Linux amd64-64-Bit Compressed References 20200804_734 (JIT enabled, AOT enabled)
OpenJ9   - 2020312e3
OMR      - 7b76a160b
JCL      - bf6e4d1c based on jdk8u265-b01)
```
```
Results for JDK=/home/mpirvu/sdks/OpenJDK8U-jre_x64_linux_openj9_2020-08-04-03-59 jvmOpts=-XX:+UseJITServer -Xjit:initialOptLevel=warm,inhibitRecompilation -Xmx256m -Xshareclasses:none
Throughput      avg=14987.82    min=14809.40    max=15123.50    stdDev=77.8     maxVar=2.12%    confInt=0.30%   samples=10
CompTime        avg=4568.40     min=4330.00     max=4666.00     stdDev=95.3     maxVar=7.76%    confInt=1.21%   samples=10
Footprint       avg=217272.40   min=215356.00   max=221312.00   stdDev=1714.0   maxVar=2.77%    confInt=0.46%   samples=10

Results for JDK=/home/mpirvu/sdks/OpenJDK8U-jre_x64_linux_openj9_2020-08-05-08-20 jvmOpts=-XX:+UseJITServer -Xjit:initialOptLevel=warm,inhibitRecompilation -Xmx256m -Xshareclasses:none
Throughput      avg=14635.48    min=14366.40    max=14774.50    stdDev=144.3    maxVar=2.84%    confInt=0.57%   samples=10
CompTime        avg=4771.00     min=4486.00     max=4893.00     stdDev=111.8    maxVar=9.07%    confInt=1.36%   samples=10
Footprint       avg=217147.60   min=215284.00   max=222072.00   stdDev=2024.0   maxVar=3.15%    confInt=0.54%   samples=10
```

CPU usage is affected as well and saw an increase of 4.4%

**Comment by mpirvu** [comment]
`Merge pull request #10118 from dmitry-ten/coalesce-methods` happened between those two builds so it's the first suspect.

**Comment by mpirvu** [comment]
PR https://github.com/eclipse/openj9/pull/10245 brings the throughput and CPU to values seen in `OpenJDK8U-jre_x64_linux_openj9_2020-08-04-03-59` SDK.
Note that #10245 is an alternative implementation of #10118, so it's not that we covered with a regression with an improvement.

**Comment by mpirvu** [comment]
Problem solved by #10245
